src/pool/mempool.rs

```diff
@@ -110,11 +94,21 @@ impl<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> AccountM
                                 .await;
                             return;
                         }
-                        let mut relayer = maybe_relayer.expect("maybe_lock is not error");
+                        let relayer = maybe_relayer.expect("not error");
 
                         // Send the Ethereum transaction using the relayer
                         let transaction_signed = transaction.to_recovered_transaction().into_signed();
-                        let res = relayer.relay_transaction(&transaction_signed).await;
+
+                        // Query the updated nonce for the account from the provider.
+                        // Via on chain query we have the most up-to-date nonce.
+                        let relayer_nonce = manager
+                            .eth_client
+                            .starknet_provider()
+                            .get_nonce(starknet::core::types::BlockId::Tag(BlockTag::Pending), relayer.address())
+                            .await
+                            .unwrap_or_default();
```

Comment:
since the relayer has access to an Starknet provider, what do you think of moving this over to the relayer?

Reply:
fixed thanks